workflow/snakemake_rules/export_for_nextstrain.smk

```diff
@@ -393,6 +417,8 @@ rule deploy_single:
     params:
         deploy_url = config["deploy_url"],
         prefix = config["auspice_json_prefix"]
+    resources:
+        concurrent_deploys = 1

```

Comment:
After looking through the [Resources section](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#resources) of Snakemake docs, I understand that standard resources such as `mem_mb` are used by Snakemake itself. However, I still don't see where custom resources such as `concurrent_deploys` are used. Can you help me understand?

Reply:
Custom resources are taken into account by Snakemake's scheduler, the same way that it takes into account CPUs and memory allocated to the workflow.

The relevant bits from the doc section you linked to are:

> [T]he scheduler will ensure that the given resources are not exceeded by running jobs.

> In general, resources are just names to the Snakemake scheduler, i.e., Snakemake does not check on the resource consumption of jobs in real time. Instead, resources are used to determine which jobs can be executed at the same time without exceeding the limits specified at the command line.

> Resources can have any arbitrary name, and must be assigned int or str values.

Reply:
So if the value of a custom resource is `int`, then the sum of those values, from all jobs where it's set, is capped by a globally defined value with the same name. Am I understanding this right? How are limits applied to `str` values?

Reply:
You're understanding it right. Numeric custom resources are used in the scheduling solutions, the ~same way the builtin CPU and memory resources are used.

A job's consumed resources are specified via the `resources:` block, `--default-resources` option, `--set-resources` option, or profile config.

A workflow's allocated resources are specified via `--resources` or `workflow.global_resources`.

The Snakemake docs aren't super clear on str-typed resources (though they may be technically accurate with a close reading). The code is clearer. While a job's consumed resources can be an int, string, or a callable, the resources allocated to the workflow can really only be ints because the scheduler needs to deal in numbers at the end of the day. The only builtin resource to take a string is `tmpdir`, and that's a specially non-scheduled resource. It's not clear to me why, or even if, it'd be useful or possible to have a job consume a custom resource that was a string, but it wouldn't be involved in the scheduler.

Reply:
Thanks, that explains it. I was confused at first because I thought there was as single `concurrent_deploys` taking the value of `1` or `2` as a default, without being used anywhere else. Now I understand that the global value is a cap on when it is set on a job level.